Azure/azure-cli
# {Compute} Fix ValueError in vm commands

**Description**  


This PR works as a part of fixing the none-actionable issues in #14986

- It fixes the ValueErrors with patterns below in VM commands.

```
not enough values to unpack (expected 4, got 2)
too many values to unpack (expected 4)
...
```
- It applies the AzCLIError in comand group
- It inroduces the new AzCLIErrorType: UsageError, ResourceNotFound 

**Testing Guide**  

**Case1**: Run `az vm image show --urn a:b:c`

before
```
UnexpectedError: The command failed with an unexpected error. Here is the traceback:
not enough values to unpack (expected 4, got 3)
Traceback (most recent call last):
  File "C:\Users\houk\Desktop\msws\env_new\lib\site-packages\knack\cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 654, in execute
    raise ex
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 718, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 711, in _run_job
    six.reraise(*sys.exc_info())
  File "C:\Users\houk\Desktop\msws\env_new\lib\site-packages\six.py", line 703, in reraise
    raise value
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 688, in _run_job
    result = cmd_copy(params)
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\__init__.py", line 325, in __call__
    return self.handler(*args, **kwargs)
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\arm.py", line 756, in handler
    show_exception_handler(ex)
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\arm.py", line 772, in show_exception_handler
    raise ex
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli-core\azure\cli\core\commands\arm.py", line 754, in handler
    return getter(**args)
  File "c:\users\houk\desktop\msws\azure-cli\src\azure-cli\azure\cli\command_modules\vm\custom.py", line 1728, in show_vm_image
    publisher, offer, sku, version = urn.split(":")
ValueError: not enough values to unpack (expected 4, got 3)
```
After
```
UsageError: --urn should be in the format of publisher:offer:sku:version
```

**Case2**: Run `az vm image show --urn a:b:c -p d`
Before
```
ValidationError: usage error: --plan STRING --offer STRING --publish STRING --version STRING | --urn STRING
```
After
```
UsageError: If using --urn, do not specify any of --publisher, --offer, --sku, --version
Try to use --urn publisher:offer:sku:version only
```
